src/Typography/Typography.component.tsx

```diff
@@ -84,40 +86,52 @@ export interface TypographyProps extends SpaceProps, React.HTMLAttributes<any> {
 
 export interface StyledTypographyProps extends TypographyProps {
     $color?: 'inherit' | 'initial' | string;
+    asVariant?: any;
+    scaleVariant?: any;
 }
 
-export const Typography = styled('span').attrs<TypographyProps>(
-    ({ color }) => ({
-        $color: color,
-        color: undefined,
-        className: 'anchor-typography',
-    })
-)<StyledTypographyProps>`
+export const Typography = styled('span').attrs<TypographyProps>(props => ({
+    $color: props.color,
+    color: undefined,
+    className: 'anchor-typography',
+    asVariant: variant({
```

Comment:
I realize the PR is already approved, but I love what you did here. Keeps the API the same but separates `as` and `asVariant` so you have options.